test/burn/org/apache/cassandra/transport/DriverBurnTest.java

```diff
@@ -77,7 +77,7 @@ public QueryMessage decode(ByteBuf body, ProtocolVersion version)
             {
                 QueryMessage queryMessage = QueryMessage.codec.decode(body, version);
                 return new QueryMessage(queryMessage.query, queryMessage.options) {
-                    protected Message.Response execute(QueryState state, long queryStartNanoTime, boolean traceRequest)
+                    public Message.Response executeSync(QueryState state, long queryStartNanoTime, boolean traceRequest)
```

Comment:
how come this gets called? does it? 

if it does it's because of some dark magic. Like - the first message on the connection (init) is always sync or sth along these lines. But this is just a hack, isn't it?
let's have a common execution path - always via, `maybeExecuteAsync`

Reply:
I don't quite get what you mean here. I think you're saying we should overwrite `maybeExecuteAsync()` here. I think it makes sense, this was an automatic refactor and I didn't pay enough attention here as it's a test and I was expecting it to fail, let me double check.

Reply:
What I mean: the `QueryMessage` is supposed to throw ISE on sync execution. So we don't expect `executeSync` to be called ever. My expectation is that overriding `executeSync` here should be a no-op.

If it isn't then why? It's so counterintuitive I'd rather avoid it even if it works.

perhaps we should override `executeAsync` and return a completed future? Overriding `maybeExecuteAsync` is probably also fine, but to me `maybeExecuteAsync` is an instance of an NVI pattern (even though everything is virtual in java...), so I'd stick with overriding `executeAsync`